src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -476,6 +476,19 @@ private void AdvanceReader(BufferSegment consumedSegment, int consumedIndex, Buf
             TrySchedule(_writerScheduler, completionData);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private long GetLength(BufferSegment startSegment, int startIndex, BufferSegment endSegment, int endIndex)
```

Comment:
should be `static` or it uses first arg for `this` making it 5 arguments and 4 is the limit for registers which pushes `endIndex` to be passed by stack. Making it static makes them all registers (the inline might fix this, but might not also)